services/src/main/java/org/keycloak/authentication/authenticators/browser/OTPFormAuthenticator.java

```diff
@@ -39,11 +41,19 @@
 public class OTPFormAuthenticator extends AbstractUsernameFormAuthenticator implements Authenticator {
     @Override
     public void action(AuthenticationFlowContext context) {
+        if (DisplayUtils.isConsole(context)) {
+            ConsoleOTPFormAuthenticator.SINGLETON.action(context);
```

Comment:
Would it not be better to obtain this from the session? With this approach it's not pluggable.

Reply:
And also so that no migration code is required.